src/test/java/core/basesyntax/service/impl/ReaderServiceImplTest.java

```diff
@@ -0,0 +1,65 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.service.ReaderService;
+import java.util.List;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ReaderServiceImplTest {
+    private static ReaderService readerService;
+
+    @BeforeClass
+    public static void beforeClass() {
+        readerService = new ReaderServiceImpl();
+    }
+
+    @Test (expected = RuntimeException.class)
+    public void reader_pathIsNull_notOk() {
+        readerService.readFromFile(null);
+    }
+
+    @Test (expected = RuntimeException.class)
+    public void reader_pathIsEmpty_notOk() {
+        readerService.readFromFile("");
+    }
+
+    @Test (expected = RuntimeException.class)
+    public void writer_pathContainsOnlySpaces_notOk() {
+        readerService.readFromFile("      ");
+    }
+
+    @Test (expected = RuntimeException.class)
+    public void reader_fileIsNotExist_notOk() {
+        readerService.readFromFile("src/test/resources/???.csv");
+    }
+
+    @Test
+    public void reader_fileIsEmpty_ok() {
+        List<String> expected = List.of();
+        assertEquals("Expected true, but was false", expected,
```

Comment:
let's maybe remove messages? they are not obligatory, and currently not very informative